### PR TITLE
fix(gtest): update `ActorMailbox::claim_value` and `ActorMailbox::reply`

### PR DESCRIPTION
**Pull request by ByteNacked**
Resolves # .

Update `ActorMailbox::claim_value` and `ActorMailbox::reply` implementation in gtest.

@reviewer-or-team

### CODE REVIEW

File: gtest/src/manager.rs [hunk 1]
```diff
@@ -294,4 +294,75 @@ impl ExtManager {
 
         Ok(message)
     }
+
+    pub(crate) fn send_reply_impl(
```
Comment (techraed) [inline gtest/src/manager.rs:298]: I'd move that to `state::mailbox::actors`, as it all related to the mailbox logic

**Review by techraed — COMMENTED** [review]
The implementation is absolutely correct.
However, some things would be very good to implement:
1. `Program::send_*` API works as extrinsics in terms of runtime. The same is about `ActorMailbox::reply*` and `ActorMailbox::claim_value`. All such user initialized messages are sent through `ExtManager::validate_and_route_dispatch`, so you need to use the former inside newly introduced `send_reply_impl` and `claim_value_impl` methods.
2. If you do the **1**, then you have to adjust the `validate_and_route_dispatch` accordingly. Please don't forget to add the check that destination is an active program.

File: gtest/src/manager.rs (hunk 1 above)
Comment (ByteNacked) [reply gtest/src/manager.rs]: I think `fn send_reply_impl` should be inside `ExtManager` to maintain a clear separation of abstraction levels.

Moved `send_reply` and `claim_value` to the `manager::exec` module, where `send` extrinsic is located.

**Comment by ByteNacked** [comment]
> * `Program::send_*` API works as extrinsics in terms of runtime. The same is about `ActorMailbox::reply*` and `ActorMailbox::claim_value`. All such user initialized messages are sent through `ExtManager::validate_and_route_dispatch`, so you need to use the former inside newly introduced `send_reply_impl` and `claim_value_impl` methods.
> 
> * If you do the **1**, then you have to adjust the `validate_and_route_dispatch` accordingly. Please don't forget to add the check that destination is an active program.

Discussed, fixed

File: gtest/src/manager.rs (hunk 1 above)
Comment (ByteNacked) [reply gtest/src/manager.rs]: Fixed

File: gtest/src/manager/exec.rs [hunk 2]
```diff
@@ -35,6 +35,33 @@ impl ExtManager {
         self.route_dispatch(dispatch)
     }
 
+    pub(crate) fn claim_value_impl(
```
Comment (techraed) [inline gtest/src/manager/exec.rs:38]: by your choice: I'd move it to `mailbox::actor` too

**Review by techraed — APPROVED** [review]

File: gtest/src/manager/exec.rs (hunk 2 above)
Comment (ByteNacked) [reply gtest/src/manager/exec.rs]: Fixed

**Pull request merged**